### DISCUSSION TITLE
tp.file.create_new doesn't work as expected

### DISCUSSION THREAD

**Discussion by mariusgerome** [SilentVoid13/Templater#1285] [Help]
Thats the template:
![ 2024-01-04 at 16 06 30](https://github.com/SilentVoid13/Templater/assets/51505161/8a8a090d-4edf-4268-9cc8-a6ca2cf8aaf9)

Thats the result:
![ 2024-01-04 at 16 07 23](https://github.com/SilentVoid13/Templater/assets/51505161/5eb046bb-467e-4c06-8605-a8c324ae535f)

I don't get it, this should be `idkyet-public: "[[2023-12-02_Sat--public]]"` with the template applied, but the newly created file has the wrong name, isn't even created, and therefore a template hasn't been applied either.

What am I doing wrong?
– M

**Comment by mariusgerome** [comment]
BTW I get the "Parsing Error" soo many times. Even when creating my absolute basic "create new note" template. Restarting Obsidian usually solves this issue – for my basic "new note" template; but not for the mentioned problem here.

**Reply by Zachatoo** [reply]
1. What is the template? Please copy/paste it here instead of screenshotting it, it's hard to read and even harder to try to reproduce since I have to write it out by hand.
2. Do you have any user scripts setup in Templater settings? Any time any template is run, Templater will check all user scripts for errors, could be an error there.

**Comment by Zachatoo** [comment]
You're trying to create a new note with the same name as the current note (and likely in the same folder). No OS I know of allows two files with the same name and extension in the same folder.

Instead of passing `tp.file.title` as the file name for the new file, you should pass `tp.file.title + "--public"`, so that the newly created file has a unique name.

---

You can check to see what the error says in the console. First open the developer tools by pressing Ctrl-Shift-I on Windows/Linux or Cmd-Opt-I on macOS, and then go to the "Console" tab.

**Reply by mariusgerome** [reply]
> Instead of passing tp.file.title as the file name for the new file, you should pass tp.file.title + "--public", so that the newly created file has a unique name.

This makes a lot of sense.

---

The console tells me two things:
![ 2024-01-04 at 16 45 26](https://github.com/SilentVoid13/Templater/assets/51505161/cd8fd4e0-bd17-4cd5-a3bd-c8cb2d328ef1)
_and I basically understand nothing_
